MeshLib/Elements/CellRule.cpp

```diff
@@ -0,0 +1,41 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "CellRule.h"
+
+#include "logog/include/logog.hpp"
+
+#include "GeoLib/AnalyticalGeometry.h"
+
+#include "MeshLib/Node.h"
+#include "Face.h"
+
+namespace MeshLib
+{
+
+const unsigned CellRule::dimension;
+
+bool CellRule::testElementNodeOrder(const Element* e)
+{
+	const MathLib::Vector3 c (e->getCenterOfGravity());
+	const unsigned nFaces (e->getNFaces());
+	for (unsigned j=0; j<nFaces; ++j)
+	{
+		MeshLib::Face const*const face (dynamic_cast<const MeshLib::Face*>(e->getFace(j)));
+		const MeshLib::Node x (*(face->getNode(1)));
```

Comment:
This is the documentation I wrote:
```c++
	 * Checks if the node order of an element is correct by testing surface normals.
	 * For 3D elements true is returned if the normals of all faces points away from the centre of 
	 * the element.
	 * Note: This method might give wrong results if something else is wrong with the element 
	 * (non-planar faces, non-convex geometry, possibly zero volume) which causes the calculated 
	 * center of gravity to lie outside of the actual element
```
I *think* I chose node 1 because that way *all* nodes are used for the test at some point, while for node 0 at least one node in every element type would be used for checking twice and one wouldn't be checked at all. (based on the definition of the ```_face_nodes``` variable)